subsys/net/conn_mgr/conn_mgr.c

```diff
@@ -334,8 +347,17 @@ static int conn_mgr_init(void)
 			CONFIG_NET_CONNECTION_MANAGER_STACK_SIZE,
 			(k_thread_entry_t)conn_mgr_handler,
 			NULL, NULL, NULL, THREAD_PRIORITY, 0, K_NO_WAIT);
+}
+
+#if defined(CONFIG_NET_CONNECTION_MANAGER_AUTO_INIT)
 
+/* Wrapper for conn_mgr_init to provide the integer return value required by SYS_INIT. */
+static int conn_mgr_auto_init(void)
+{
+	conn_mgr_init();
 	return 0;
 }
 
-SYS_INIT(conn_mgr_init, APPLICATION, CONFIG_NET_CONNECTION_MANAGER_PRIORITY);
+SYS_INIT(conn_mgr_auto_init, APPLICATION, CONFIG_NET_CONNECTION_MANAGER_AUTO_INIT_LEVEL);
```

Comment:
Rather than make the `SYS_INIT` call optional, let's have the `conn_mgr_thread` fully handle initialization, and pend an "all clear" signal.

That way it isn't possible for applications to accidentially initialize `conn_mgr` ***sooner*** than it normally would.

Reply:
Do I understand it correctly, that `SYS_INIT()` would spawn the thread (unconditionally) which would then block on a semaphore for instance, till `conn_mgr_init();` is called?

Reply:
Yes, precisely -- although I have decided to skip this feature for this PR